app.py

```diff
@@ -140,13 +153,61 @@ def handle_message(payload):
                     text=f"Error sending message: {response['error']}",
                 )
                 logging.error(f"Error sending message: {response['error']}")
+    # if message.get("channel_type") == "im":
+    #     user = message["user"]  # The user ID of the person who sent the message
+    #     text = message.get("text", "")  # The text of the message
+    #     try:
+    #         if message.get("user") != bot_user_id:
+    #             client.chat_postMessage(channel=JOINS_CHANNEL_ID, text=f"<@{user}> said {text}")
+    #         # Respond to the direct message
+    #         client.chat_postMessage(channel=user, text=f"Hello <@{user}>, you said: {text}")
+    #     except SlackApiError as e:
+    #         print(f"Error sending response: {e.response['error']}")
+
+
+@slack_events_adapter.on("message")
+def gpt_bot(payload):
+    token_limit = 1000
+    token_per_prompt = 100
+    user = "D078YQ93TSL"
+    message = payload.get("event", {})
+
     if message.get("channel_type") == "im":
-        user = message["user"]  # The user ID of the person who sent the message
-        text = message.get("text", "")  # The text of the message
+        doubt = message.get("text", "")
+        prompt = template.format(doubt=doubt)
+
+        today = datetime.now(timezone.utc).date()
+        rate_limit_key = f"global_daily_request_{today}"
+        total_token_used = cache.get(rate_limit_key, 0)
+
+        if len(doubt) > 50:
```

Comment:
To be consistent w/ the error message:
```suggestion
        if len(doubt) >= 50:
```

Also consider not using a magic number -- create a descriptively named variable for that.

Reply:
Sure!! thanks for the suggestion